### DISCUSSION TITLE
Does URLS_ERROR provide info on which OAuth2.0 client or RP may have triggered the error?

### DISCUSSION THREAD

**Discussion by mig5** [ory/hydra#3313] [Q&A] [answered]
I am making a dedicated error page on my Login and Consent app for handling the `URLS_ERROR` route.

I know that the redirect to `URLS_ERROR` may contain `error` and `error_description` query parameters to help explain the issue.

Is there any way it can tell me which RP made the request that resulted in the error? I know that some other library functions like the Login and Consent ones can provide a redirect_to URL (or similarly named) in the event of an error e.g error with the challenge tokens.

My use case is that I want to provide a button on the error page saying 'Go back' or similar, to help the user find their way back to the app they were on.

Side topic: Displaying the error/error_description info also feels a bit icky because they are query parameters, so technically 'attacker controlled input'. I wonder if there's a way I can make my app obtain the error/error_description from a Hydra API endpoint rather than trust what's in the query parameters?

Thanks!

**Comment by aeneasr** [comment, marked as answer]
This is unfortunately not possible, the query parameters you have in the URL are currently the only ones.
